src/shared/services/bookmarks-views-plays-service.ts

```diff
@@ -0,0 +1,370 @@
+type EventContentType = EventContentTypeSimplified | 'bundle';
+
+interface QueryDefinition {
+	query: any;
+	variables: ((uuid: string, profileId: string) => any) | ((uuid: string) => any);
+}
+
+const EVENT_QUERIES: {
+	[action in EventActionExtended]: { [contentType in EventContentTypeSimplified]: QueryDefinition };
+} = {
+	bookmark: {
+		item: {
+			query: INSERT_ITEM_BOOKMARK,
+			variables: (itemUuid: string, profileId: string) => ({
+				bookmarkItem: {
+					item_id: itemUuid,
+					profile_id: profileId,
+				},
+			}),
+		},
+		collection: {
+			query: INSERT_COLLECTION_BOOKMARK,
+			variables: (collectionUuid: string, profileId: string) => ({
+				bookmarkItem: {
+					collection_uuid: collectionUuid,
+					profile_id: profileId,
+				},
+			}),
+		},
+	},
+	unbookmark: {
+		item: {
+			query: REMOVE_ITEM_BOOKMARK,
+			variables: (itemUuid: string, profileId: string) => ({
+				itemUuid,
+				profileId,
+			}),
+		},
+		collection: {
+			query: REMOVE_COLLECTION_BOOKMARK,
+			variables: (collectionUuid: string, profileId: string) => ({
+				collectionUuid,
+				profileId,
+			}),
+		},
+	},
+	view: {
+		item: {
+			query: INCREMENT_ITEM_VIEWS,
+			variables: (itemUuid: string) => ({
+				itemUuid,
+			}),
+		},
+		collection: {
+			query: INCREMENT_COLLECTION_VIEWS,
+			variables: (collectionUuid: string) => ({
+				collectionUuid,
+			}),
+		},
+	},
+	play: {
+		item: {
+			query: INCREMENT_ITEM_PLAYS,
+			variables: (itemUuid: string) => ({
+				itemUuid,
+			}),
+		},
+		collection: {
+			query: INCREMENT_COLLECTION_PLAYS,
+			variables: (collectionUuid: string) => ({
+				collectionUuid,
+			}),
+		},
+	},
+	viewInit: {
+		item: {
+			query: INIT_ITEM_VIEWS,
+			variables: (itemUuid: string) => ({
+				itemUuid,
+			}),
+		},
+		collection: {
+			query: INIT_COLLECTION_VIEWS,
+			variables: (collectionUuid: string) => ({
+				collectionUuid,
+			}),
+		},
+	},
+	playInit: {
+		item: {
+			query: INIT_ITEM_PLAYS,
+			variables: (itemUuid: string) => ({
+				itemUuid,
+			}),
+		},
+		collection: {
+			query: INIT_COLLECTION_PLAYS,
+			variables: (collectionUuid: string) => ({
+				collectionUuid,
+			}),
+		},
+	},
+};
+
+export class BookmarksViewsPlaysService {
+	public static async action(
+		action: EventAction,
+		contentType: EventContentType,
+		contentUuid: string,
+		user?: Avo.User.User,
+		silent: boolean = true
+	): Promise<void> {
+		try {
+			// bundle is handled the same way as a collection
+			const contentTypeSimplified = contentType === 'bundle' ? 'collection' : contentType;
+
+			const mutation = get(EVENT_QUERIES, [action, contentTypeSimplified, 'query']);
+			const variables = get(EVENT_QUERIES, [action, contentTypeSimplified, 'variables'], () => {})(
+				contentUuid,
+				get(user, 'profile.id')
+			);
+			if (!mutation || !variables) {
+				throw new CustomError('Failed to find query/variables in query lookup table');
+			}
+			console.log('query', print(mutation));
+			console.log('variables', variables);
```

Comment:
Hanz, you missed a spot.

Reply:
fixed in: feature/bookmarks-in-workspace